framework/src/modules/poa/commands/update_authority.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { MAX_UINT64, validator } from '@liskhq/lisk-validator';
+import { address, bls } from '@liskhq/lisk-cryptography';
+import { codec } from '@liskhq/lisk-codec';
+import { objects as objectUtils } from '@liskhq/lisk-utils';
+import { BaseCommand } from '../../base_command';
+import { updateAuthorityValidatorSchema, validatorSignatureMessageSchema } from '../schemas';
+import {
+	COMMAND_UPDATE_AUTHORITY,
+	MAX_NUM_VALIDATORS,
+	MESSAGE_TAG_POA,
+	EMPTY_BYTES,
+	UpdateAuthority,
+} from '../constants';
+import {
+	CommandExecuteContext,
+	CommandVerifyContext,
+	VerificationResult,
+	VerifyStatus,
+} from '../../../state_machine';
+import { UpdateAuthorityValidatorParams } from '../types';
+import { ChainPropertiesStore, SnapshotStore, ValidatorStore } from '../stores';
+import { ValidatorsMethod } from '../../pos/types';
+import { AuthorityUpdateEvent } from '../events/authority_update';
+
+export class UpdateAuthorityCommand extends BaseCommand {
+	public schema = updateAuthorityValidatorSchema;
+	private _validatorsMethod!: ValidatorsMethod;
+
+	public get name(): string {
+		return COMMAND_UPDATE_AUTHORITY;
+	}
+
+	public addDependencies(validatorsMethod: ValidatorsMethod) {
+		this._validatorsMethod = validatorsMethod;
+	}
+
+	public async verify(
+		context: CommandVerifyContext<UpdateAuthorityValidatorParams>,
+	): Promise<VerificationResult> {
+		const { newValidators, threshold, validatorsUpdateNonce } = context.params;
+		try {
+			validator.validate(updateAuthorityValidatorSchema, context.params);
+		} catch (err) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: err as Error,
+			};
+		}
+		if (newValidators.length < 1 || newValidators.length > MAX_NUM_VALIDATORS) {
+			throw new Error('Invalid number of newValidators.');
+		}
+		if (
+			!objectUtils.bufferArrayOrderByLex(newValidators.map(newValidator => newValidator.address))
+		) {
+			throw new Error('Addresses in newValidators are not lexicographical ordered.');
+		}
+
+		let totalWeight = BigInt(0);
+		for (const newValidator of newValidators) {
+			const validatorExists = await this.stores
+				.get(ValidatorStore)
+				.has(context, newValidator.address);
+			if (!validatorExists) {
+				throw new Error(
+					`${address.getLisk32AddressFromPublicKey(
+						newValidator.address,
+					)} is not a valid validator.`,
+				);
+			}
+			totalWeight += newValidator.weight;
+		}
+
+		if (totalWeight > MAX_UINT64) {
+			throw new Error('TotalWeight out of range.');
```

Comment:
maybe we can give more informative error messages
```suggestion
			throw new Error('TotalWeight should be less than MAX_UINT64.');
```

Reply:
Updated on https://github.com/LiskHQ/lisk-sdk/pull/8527